zip-0243.rst

```diff
@@ -0,0 +1,291 @@
+::
+
+  ZIP: 243
+  Title: Transaction Signature Verification for Sapling
+  Author: Jack Grigg <jack@z.cash>
+          Daira Hopwood <daira@z.cash>
+  Category: Consensus
+  Created: 2018-04-10
+  License: MIT
+
+
+Terminology
+===========
+
+The key words "MUST" and "MUST NOT" in this document are to be interpreted as described in RFC 2119.
+
+The terms "branch" and "network upgrade" in this document are to be interpreted as described in ZIP 200. [#zip-0200]_
+
+The term "Overwinter" in this document is to be interpreted as described in ZIP 201. [#zip-0201]_
+
+The term "Sapling" in this document is to be interpreted as described in ZIP XXX. [#zip-0XXX]_
+
+
+Abstract
+========
+
+This proposal defines a new transaction digest algorithm for signature verification from the Sapling network
+upgrade, to account for the presence of Sapling shielded inputs and outputs in transactions.
+
+
+Motivation
+==========
+
+The Sapling network upgrade introduced new shielded inputs and outputs. We want these to be covered by the
+transaction digest algorithm used for signatures, in order to ensure they are correctly bound.
+
+
+Specification
+=============
+
+A new transaction digest algorithm is defined::
+
+  BLAKE2b-256 hash of the serialization of:
+    1. header of the transaction (4-byte little endian)
+    2. nVersionGroupId of the transaction (4-byte little endian)
+    3. hashPrevouts (32-byte hash)
+    4. hashSequence (32-byte hash)
+    5. hashOutputs (32-byte hash)
+    6. hashJoinSplits (32-byte hash)
+    7. hashShieldedSpends (32-byte hash)
+    8. hashShieldedOutputs (32-byte hash)
+    9. nLockTime of the transaction (4-byte little endian)
+   10. nExpiryHeight of the transaction (4-byte little endian)
+   11. valueBalance of the transaction (8-byte little endian)
+   12. sighash type of the signature (4-byte little endian)
+
+   13. If we are serializing an input (i.e. this is not a JoinSplit signature hash):
```

Comment:
If we are serializing a transparent input (i.e. this hash is not for a JoinSplit signature, Spend authorization signature, or binding signature):